java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -1244,6 +1246,21 @@ public CompletableFuture<String> xadd(
         return commandManager.submitNewCommand(XAdd, arguments, this::handleStringOrNullResponse);
     }
 
+    @Override
+    public CompletableFuture<Map<String, Map<String, Object[][]>>> xread(
+            @NonNull Map<String, String> keysAndIds) {
+        return xread(keysAndIds, StreamReadOptions.builder().build());
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public CompletableFuture<Map<String, Map<String, Object[][]>>> xread(
+            @NonNull Map<String, String> keysAndIds, @NonNull StreamReadOptions options) {
+        String[] arguments = options.toArgs(keysAndIds);
+        return commandManager.submitNewCommand(
+                XRead, arguments, response -> handleMapOrNullResponse(response));
```

Comment:
I prefer having custom handle function which casts this `[][]` nested into maps to return `Map<String, Map<String, String[][]>>`

Reply:
I misunderstood.  I thought you wanted it removed.